src/lib/blueprint.ts

```diff
@@ -941,6 +948,25 @@ const createResponse = (
   }
 }
 
+const validateActionAttemptType = (
+  actionAttemptType: string | undefined,
+  responseKey: string,
+  path: string,
+): string | undefined => {
+  const excludedPaths = ['/action_attempts']
+  const isPathExcluded = excludedPaths.some((p) => path.startsWith(p))
+
+  if (
+    actionAttemptType == null &&
+    responseKey === 'action_attempt' &&
+    !isPathExcluded
+  ) {
+    throw new Error(`Missing action_attempt_type for path ${path}`)
```

Comment:
Can we also validate the action attempt type is a valid one?

Reply:
976a20fa86c60872b6772c1ae450bb1a22877ffb